src/charm.py

```diff
@@ -112,7 +112,11 @@ def _on_get_issued_certificates(self, event: ActionEvent) -> None:
         if not certificates:
             event.fail("No certificates issued yet.")
             return
-        results = {"certificates": json.dumps([vars(certificate) for certificate in certificates])}
+        results = {
+            "certificates": json.dumps(
```

Comment:
wouldn't 
```
"certificates": [certificate.to_json() for certificate in certificates]
```
work?

I'm not sure using `json.loads` is safe here, dataclasses should not be considered serialisable.

Reply:
nor deserialisable

Reply:
Yes we can do this, ok nice :)